cli/azd/internal/vsrpc/environment_service.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package vsrpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+)
+
+// environmentService is the RPC server for the '/environmentService/v1.0' endpoint.
+type environmentService struct {
+	server *Server
+}
+
+func newEnvironmentService(server *Server) *environmentService {
+	return &environmentService{
+		server: server,
+	}
+}
+
+// GetEnvironmentsAsync is the server implementation of:
+// ValueTask<IEnumerable<EnvironmentInfo>> GetEnvironmentsAsync(Session, IObserver<ProgressMessage>, CancellationToken);
+func (s *environmentService) GetEnvironmentsAsync(
+	ctx context.Context, sessionId Session, observer IObserver[ProgressMessage],
+) ([]*EnvironmentInfo, error) {
+	session, err := s.server.validateSession(ctx, sessionId)
+	if err != nil {
+		return nil, err
+	}
+
+	session.sessionMu.Lock()
+	defer session.sessionMu.Unlock()
+
+	var c struct {
+		envManager environment.Manager `container:"type"`
+	}
+
+	if err := session.container.Fill(&c); err != nil {
+		return nil, err
+	}
+
+	envs, err := c.envManager.List(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("listing environments: %w", err)
+	}
+
+	infos := make([]*EnvironmentInfo, len(envs))
+	for i, env := range envs {
+		infos[i] = &EnvironmentInfo{
+			Name:      env.Name,
+			IsCurrent: env.IsDefault,
+		}
+	}
+
+	return infos, nil
+}
+
+// SetCurrentEnvironmentAsync is the server implementation of:
+// ValueTask<bool> SetCurrentEnvironmentAsync(Session, string, IObserver<ProgressMessage>, CancellationToken);
+func (s *environmentService) SetCurrentEnvironmentAsync(
+	ctx context.Context, sessionId Session, name string, observer IObserver[ProgressMessage],
+) (bool, error) {
+	session, err := s.server.validateSession(ctx, sessionId)
+	if err != nil {
+		return false, err
+	}
+
+	session.sessionMu.Lock()
+	defer session.sessionMu.Unlock()
+
+	var c struct {
+		azdCtx *azdcontext.AzdContext `container:"type"`
+	}
+
+	if err := session.container.Fill(&c); err != nil {
+		return false, err
+	}
+
+	if err := c.azdCtx.SetDefaultEnvironmentName(name); err != nil {
+		return false, fmt.Errorf("saving default environment: %w", err)
+	}
+
+	return true, nil
+}
+
+// DeleteEnvironmentAsync is the server implementation of:
+// ValueTask<bool> DeleteEnvironmentAsync(Session, string, IObserver<ProgressMessage>, CancellationToken);
+func (s *environmentService) DeleteEnvironmentAsync(
+	ctx context.Context, sessionId Session, name string, observer IObserver[ProgressMessage],
+) (bool, error) {
+	session, err := s.server.validateSession(ctx, sessionId)
+	if err != nil {
+		return false, err
+	}
+
+	session.sessionMu.Lock()
+	defer session.sessionMu.Unlock()
+
+	// TODO(ellismg): Implement this.
+	return false, errors.New("not implemented")
```

Comment:
https://github.com/Azure/azure-dev/issues/937

There's many questions we don't have an answer for when it comes to deleting environment....  like
- Should we call azd down or just leak resources?
- Should we allow restore and just mark environment as deleted? (best practice)
- Can we delete the current default environment?  should we jump to the next available env as new default? or ask the user?
- What if I delete env foo and then create env foo again?  (what happen on Azure?)

.. etc etc..

Reply:
> There's many questions we don't have an answer for when it comes to deleting environment.... like

Yup - agree we need to close on all of these - I do think implementing this RPC will be easy (we will just call `envManager.Delete()` once we have it) but we should start thing about the answers to the above and coming up with what we want to do about `azd env delete` as well.

Reply:
Capture this excellent feedback and some of my thoughts in #3285.  We can work with the VS folks to figure out what we want to do here and what we want the UX to be.  Suspect once we have the design it will be easy to write the code.